FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -2370,6 +2370,114 @@ \subsection{Evaluation}
+
+\subsection{Tree Construction Rules}
+
+\begin{enumerate}
+    \item If only positive or negative instances are left, stop branching and assign the corresponding decision as leaf node\\Example: On Offer = 0\% or 50\%
+    \item If some positive and some negative examples remain, choose another feature and refine the tree one step further\\Example: On Offer = 20\% $\rightarrow$ next feature: Saturday
+    \item If no instances are left, such a combination of feature values does not occur in the training set. In this case, look at the parent node and decide according to the more frequent label
+    \item If there are still instances but no features any more, the training data s contradictory, noisy, non-deterministic or contains hidden features. In his case, decide according to the more frequent label.
+\end{enumerate}
+
+\subsubsection{Splitting Criterion}
+We need a way to measure how informative a feature is. Then, we always select the most informative feature. There are two main variants called information gain and Gini index.
+
+\subsection{Gini Impurity}
+Gini impurity is a measure of how often a randomly chosen element from the set would be incorrectly labeled if it was randomly labeled according to the distribution of labels in the subset. Gini impurity is non-negative and reaches zero when all cases in the node fall into a single target category. Choose the feature with lowest Gini impurity value for splitting.
+
+\begin{equation}
+    \sum_{i=1}^{J}p_i \sum_{k\neq i} p_k = \sum_{i=1}^{J}p_i(1-p_i) = \sum_{i=1}^{J}(p_i - p_i^2) = \sum_{i=1}^{J}p_i - \sum_{i=1}^{J}p_i^2 = 1 -\sum_{i=1}^{J}p_i^2
+\end{equation}
+
+\subsubsection{Example for a Gini calculation}
+\begin{itemize}[leftmargin=*, labelindent=4cm, labelsep=1cm]
+    \item[$p(\text{Weather = Rain})$] $\frac{7}{12}$
+    \item[$p(\text{Weather = Sun})$] $\frac{3}{12}$
+    \item[$p(\text{Weather = Cloudy})$] $\frac{2}{12}$
+\end{itemize}
+
+\begin{equation*}
+    \text{Impurity(Weather)} = 1 - (\frac{7}{12})^2 - (\frac{3}{12})^2 - (\frac{2}{12})^2 = \frac{41}{72} \approx 0.569
+\end{equation*}
+
+\subsection{Regression Trees in CART}
+
+The regression tress predict the average value of all instances in a leaf node and are grown like classification trees but with a different splitting criteria. As splitting criteria, they minimise the variance of the values in the same subset.
+
+\begin{figure}[tbh!]
+    \centering
+    \includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/regression_tree_CART}
+    \caption{A CART regression tree}
+    \label{fig:regressiontreecart}
+\end{figure}
+
+\subsection{Advantages and Disadvantages of Decision Trees}
+
+\textbf{Advantages}
+\begin{itemize}
+    \item Decision trees are simple to understand and interpret for humans
+    \item Decision trees can handle both numeric and categorical data
+    \item Decision trees require almost no data preparation
+    \item Decision trees perform well even on larger datasets
+\end{itemize}
```

Comment:
- Simple to understand [...]
- Can handle both [...]
- Require almost no [...]
- Perform well [...]

Leave the Decision tree out.